### PR TITLE
Update WebRender

### LINKED ISSUE #17176: reddit.com: Assertion failed: self.scroll_root_parents.contains_key(&scroll_root_id)
called `Result::unwrap()` on an `Err` value: "SendError(..)"
### Servo Version:
Servo 0.0.1-c260c6e
### Backtrace:
```

WARNING: <Constellation>: creating replacement pipeline for about:failure
WARNING: <ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }>: layout thread unreachable - leaking layout data
WARNING: <ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }>: layout thread unreachable - leaking layout data
WARNING: <ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(1) }>: layout thread unreachable - leaking layout data
WARNING: <ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(2) }>: Network error: Internal("Unexpected scheme")
WARNING: <ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(2) }>: Resuming an already resumed timer.
WARNING: <ScriptThread PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(2) }>: stop_parsing not implemented, full speed ahead!
WARNING: <ScriptThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(6) }>: Panic hook called.
ERROR: called `Result::unwrap()` on an `Err` value: "SendError(..)"

stack backtrace:
   0:        0x1105f13fe - backtrace::backtrace::trace::h15d769766aa71ae0
   1:        0x1105f158c - backtrace::capture::Backtrace::new::hde1afc496271e12d
   2:        0x10f6eeb29 - constellation::constellation::log_entry::hc60ffaf2b8a0de43
   3:        0x10f6edb42 - _$LT$constellation..constellation..FromCompositorLogger$u20$as$u20$log..Log$GT$::log::hb085a8ba70ec667a
   4:        0x11082d3ae - log::__log::h9babd36d99ae110e
   5:        0x10eb53c40 - servo::main::_$u7b$$u7b$closure$u7d$$u7d$::h2d732856f9776e1f
   6:        0x11087d696 - std::panicking::rust_panic_with_hook::h8b9b25777425677b
   7:        0x11087d4f4 - std::panicking::begin_panic::h69fe06a9d67b6165
   8:        0x11087d462 - std::panicking::begin_panic_fmt::haf8695bfb6c52436
   9:        0x11087d3c7 - rust_begin_unwind
  10:        0x1108a8d00 - core::panicking::panic_fmt::hfedd6515bcee8bb1
  11:        0x10ef6d1b0 - core::result::unwrap_failed::he02056d3c716db25
  12:        0x10f2306f4 - script::dom::window::Window::force_reflow::h931034571b5464d0
  13:        0x10f29d124 - script::script_thread::ScriptThread::handle_event::h221932cb36c48172
  14:        0x10f27a908 - script::script_thread::ScriptThread::handle_msgs::had90faa4ac3794d3
  15:        0x10ed94ff7 - std::sys_common::backtrace::__rust_begin_short_backtrace::hbc61873337296b9e
  16:        0x10edb0c7d - std::panicking::try::do_call::h1340531f44a9a1be
  17:        0x11087e7ca - __rust_maybe_catch_panic
  18:        0x10ef6dc66 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::h3a90dd4faf2ab5d0
  19:        0x11087a565 - std::sys::imp::thread::Thread::new::thread_start::h993b7e5cb7871f66
  20:     0x7fff926af93a - _pthread_body
  21:     0x7fff926af886 - _pthread_start
```
This report was generated by the browser.html issue reporter.

### LINKED ISSUE #19287: assertion failed: !self.nodes.contains_key(&id)
This is the same panic as #18441, but probably with a different cause.

Found with modified domato

Testcase:
```html
<details open style="overflow: auto">
```

```gdb
assertion failed: !self.nodes.contains_key(&id) (thread RenderBackend, at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/clip_scroll_tree.rs:493)
stack backtrace:
   0:     0x55b6a85741b4 - backtrace::backtrace::libunwind::trace
                        at .cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/libunwind.rs:53
                         - backtrace::backtrace::trace<closure>
                        at .cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/mod.rs:42
   1:     0x55b6a856950f - backtrace::capture::{{impl}}::new
                        at .cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/capture.rs:64
   2:     0x55b6a7a725de - servo::main::{{closure}}
                        at ports/servo/main.rs:130
   3:     0x55b6ae2b7556 - std::panicking::rust_panic_with_hook
                        at /checkout/src/libstd/panicking.rs:578
   4:     0x55b6a8b85d96 - std::panicking::begin_panic<&str>
                        at /checkout/src/libstd/panicking.rs:538
   5:     0x55b6a89126f6 - webrender::clip_scroll_tree::{{impl}}::add_node
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/clip_scroll_tree.rs:493
   6:     0x55b6a8b3e7d3 - webrender::frame_builder::{{impl}}::add_scroll_frame
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/frame_builder.rs:502
   7:     0x55b6a8b34f4e - webrender::frame::{{impl}}::flatten_scroll_frame
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/frame.rs:197
   8:     0x55b6a8b3795a - webrender::frame::{{impl}}::flatten_item
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/frame.rs:586
   9:     0x55b6a8b3499c - webrender::frame::{{impl}}::flatten_items
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/frame.rs:151
  10:     0x55b6a8b35827 - webrender::frame::{{impl}}::flatten_stacking_context
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/frame.rs:294
  11:     0x55b6a8b374c2 - webrender::frame::{{impl}}::flatten_item
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/frame.rs:534
  12:     0x55b6a8b3499c - webrender::frame::{{impl}}::flatten_items
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/frame.rs:151
  13:     0x55b6a8b34675 - webrender::frame::{{impl}}::flatten_root
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/frame.rs:117
  14:     0x55b6a8b3aa13 - webrender::frame::{{impl}}::create
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/frame.rs:1121
  15:     0x55b6a89f3652 - webrender::render_backend::{{impl}}::build_scene
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/render_backend.rs:89
  16:     0x55b6a89f4c34 - webrender::render_backend::{{impl}}::process_document
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/render_backend.rs:293
  17:     0x55b6a89f60f5 - webrender::render_backend::{{impl}}::run
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/render_backend.rs:502
  18:     0x55b6a8958425 - webrender::renderer::{{impl}}::new::{{closure}}
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/aa74cdf/webrender/src/renderer.rs:1815
  19:     0x55b6a889b05a - std::sys_common::backtrace::__rust_begin_short_backtrace<closure,()>
                        at /checkout/src/libstd/sys_common/backtrace.rs:134
  20:     0x55b6a8a89cb3 - std::thread::{{impl}}::spawn::{{closure}}::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:400
  21:     0x55b6a88842ca - std::panic::{{impl}}::call_once<(),closure>
                        at /checkout/src/libstd/panic.rs:296
  22:     0x55b6a8b870d2 - std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panicking.rs:480
  23:     0x55b6ae2be69c - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
  24:     0x55b6a8b862a2 - std::panicking::try<(),std::panic::AssertUnwindSafe<closure>>
                        at /checkout/src/libstd/panicking.rs:459
  25:     0x55b6a889c235 - std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panic.rs:361
  26:     0x55b6a8a89ad1 - std::thread::{{impl}}::spawn::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:399
  27:     0x55b6a8ab4dc3 - alloc::boxed::{{impl}}::call_box<(),closure>
                        at /checkout/src/liballoc/boxed.rs:762
  28:     0x55b6ae2b626b - alloc::boxed::{{impl}}::call_once<(),()>
                        at /checkout/src/liballoc/boxed.rs:772
                         - std::sys_common::thread::start_thread
                        at /checkout/src/libstd/sys_common/thread.rs:24
                         - std::sys::imp::thread::{{impl}}::new::thread_start
                        at /checkout/src/libstd/sys/unix/thread.rs:90
  29:     0x7fa5e732c493 - start_thread
  30:     0x7fa5e6c51abe - __clone
  31:                0x0 - <unknown>
ERROR:servo: assertion failed: !self.nodes.contains_key(&id)
```

### LINKED ISSUE #19648: crash on ebay.de: WR assertion failed: !self.nodes.contains_key(&id) under flatten_scroll_frame.
Shows blank window (but w/ correct title) for over a minute and then crashes.

Maybe even two separate issues: a) blank while waiting for network b) crashing on network error.

```
(jessie64)nekrad@orion:~/src/moz/servo$ ./mach run -d http://www.ebay.de/
assertion failed: !self.nodes.contains_key(&id) (thread WRRenderBackend#0, at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/clip_scroll_tree.rs:520)
stack backtrace:
   0:     0x55a4d827d7bc - backtrace::backtrace::libunwind::trace
                        at .cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/libunwind.rs:53
                         - backtrace::backtrace::trace<closure>
                        at .cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/mod.rs:42
   1:     0x55a4d827d85f - backtrace::capture::{{impl}}::new
                        at .cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/capture.rs:64
   2:     0x55a4d24d5662 - servo::main::{{closure}}
                        at ports/servo/main.rs:130
   3:     0x55a4d82ba3b6 - std::panicking::rust_panic_with_hook
                        at /checkout/src/libstd/panicking.rs:578
   4:     0x55a4d604442d - std::panicking::begin_panic<&str>
                        at /checkout/src/libstd/panicking.rs:538
   5:     0x55a4d5e1e14b - webrender::clip_scroll_tree::{{impl}}::add_node
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/clip_scroll_tree.rs:520
   6:     0x55a4d60c717a - webrender::frame_builder::{{impl}}::add_scroll_frame
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame_builder.rs:704
   7:     0x55a4d5e31430 - webrender::frame::{{impl}}::flatten_scroll_frame
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:203
   8:     0x55a4d5e33c81 - webrender::frame::{{impl}}::flatten_item
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:579
   9:     0x55a4d5e31028 - webrender::frame::{{impl}}::flatten_items
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:153
  10:     0x55a4d5e31c15 - webrender::frame::{{impl}}::flatten_stacking_context
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:291
  11:     0x55a4d5e337e0 - webrender::frame::{{impl}}::flatten_item
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:527
  12:     0x55a4d5e31028 - webrender::frame::{{impl}}::flatten_items
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:153
  13:     0x55a4d5e31c15 - webrender::frame::{{impl}}::flatten_stacking_context
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:291
  14:     0x55a4d5e337e0 - webrender::frame::{{impl}}::flatten_item
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:527
  15:     0x55a4d5e31028 - webrender::frame::{{impl}}::flatten_items
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:153
  16:     0x55a4d5e30cc7 - webrender::frame::{{impl}}::flatten_root
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:116
  17:     0x55a4d5e36437 - webrender::frame::{{impl}}::create
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/frame.rs:1054
  18:     0x55a4d5e3f370 - webrender::render_backend::{{impl}}::build_scene
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/render_backend.rs:94
  19:     0x55a4d5e406da - webrender::render_backend::{{impl}}::process_document
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/render_backend.rs:292
  20:     0x55a4d5e41af6 - webrender::render_backend::{{impl}}::run
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/render_backend.rs:491
  21:     0x55a4d5f27db8 - webrender::renderer::{{impl}}::new::{{closure}}
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender/src/renderer.rs:2208
  22:     0x55a4d5d9cc64 - std::sys_common::backtrace::__rust_begin_short_backtrace<closure,()>
                        at /checkout/src/libstd/sys_common/backtrace.rs:133
  23:     0x55a4d604438b - std::thread::{{impl}}::spawn::{{closure}}::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:406
  24:     0x55a4d5d8ebc4 - std::panic::{{impl}}::call_once<(),closure>
                        at /checkout/src/libstd/panic.rs:300
  25:     0x55a4d6045794 - std::panicking::try::do_call<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panicking.rs:480
  26:     0x55a4d82c2b5e - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
  27:     0x55a4d604484c - std::panicking::try<(),std::panic::AssertUnwindSafe<closure>>
                        at /checkout/src/libstd/panicking.rs:459
  28:     0x55a4d5d9f7ec - std::panic::catch_unwind<std::panic::AssertUnwindSafe<closure>,()>
                        at /checkout/src/libstd/panic.rs:365
  29:     0x55a4d60441dd - std::thread::{{impl}}::spawn::{{closure}}<closure,()>
                        at /checkout/src/libstd/thread/mod.rs:405
  30:     0x55a4d6081775 - alloc::boxed::{{impl}}::call_box<(),closure>
                        at /checkout/src/liballoc/boxed.rs:762
  31:     0x55a4d82b901b - alloc::boxed::{{impl}}::call_once<(),()>
                        at /checkout/src/libstd/sys_common/thread.rs:23
                         - std::sys::imp::thread::{{impl}}::new::thread_start
                        at /checkout/src/libstd/sys/unix/thread.rs:90
  32:     0x7fb674aa1063 - start_thread
  33:     0x7fb67391e62c - clone
  34:                0x0 - <unknown>
ERROR:servo: assertion failed: !self.nodes.contains_key(&id)
called `Result::unwrap()` on an `Err` value: Io(Error { repr: Os { code: 104, message: "Connection reset by peer" } }) (thread main, at /checkout/src/libcore/result.rs:906)
stack backtrace:
   0:     0x55a4d827d7bc - backtrace::backtrace::libunwind::trace
                        at .cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/libunwind.rs:53
                         - backtrace::backtrace::trace<closure>
                        at .cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/backtrace/mod.rs:42
   1:     0x55a4d827d85f - backtrace::capture::{{impl}}::new
                        at .cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.2/src/capture.rs:64
   2:     0x55a4d24d5662 - servo::main::{{closure}}
                        at ports/servo/main.rs:130
   3:     0x55a4d82ba3b6 - std::panicking::rust_panic_with_hook
                        at /checkout/src/libstd/panicking.rs:578
   4:     0x55a4d82ba24c - std::panicking::begin_panic<alloc::string::String>
                        at /checkout/src/libstd/panicking.rs:538
   5:     0x55a4d82ba169 - std::panicking::begin_panic_fmt
                        at /checkout/src/libstd/panicking.rs:522
   6:     0x55a4d82ba0f2 - std::panicking::rust_begin_panic
                        at /checkout/src/libstd/panicking.rs:498
   7:     0x55a4d82ca910 - core::panicking::panic_fmt
                        at /checkout/src/libcore/panicking.rs:71
   8:     0x55a4d7782617 - core::result::unwrap_failed<alloc::boxed::Box<bincode::internal::ErrorKind>>
                        at /checkout/src/libcore/macros.rs:23
   9:     0x55a4d777fa52 - core::result::{{impl}}::unwrap<webrender_api::api::HitTestResult,alloc::boxed::Box<bincode::internal::ErrorKind>>
                        at /checkout/src/libcore/result.rs:772
  10:     0x55a4d775c82a - webrender_api::api::{{impl}}::hit_test
                        at .cargo/git/checkouts/webrender-c3596abe1cf4f320/1142dfc/webrender_api/src/api.rs:696
  11:     0x55a4d24b4a0b - compositing::compositor::{{impl}}::hit_test_at_point<glutin_app::window::Window>
                        at /home/nekrad/src/moz/servo/components/compositing/compositor.rs:735
  12:     0x55a4d24bd086 - compositing::compositor::{{impl}}::dispatch_mouse_window_move_event_class<glutin_app::window::Window>
                        at /home/nekrad/src/moz/servo/components/compositing/compositor.rs:762
  13:     0x55a4d24bc543 - compositing::compositor::{{impl}}::on_mouse_window_move_event_class<glutin_app::window::Window>
                        at /home/nekrad/src/moz/servo/components/compositing/compositor.rs:750
  14:     0x55a4d2496897 - servo::{{impl}}::handle_window_event<glutin_app::window::Window>
                        at /home/nekrad/src/moz/servo/components/servo/lib.rs:292
  15:     0x55a4d2494249 - servo::{{impl}}::handle_events<glutin_app::window::Window>
                        at /home/nekrad/src/moz/servo/components/servo/lib.rs:481
  16:     0x55a4d24d4bb5 - servo::main
                        at ports/servo/main.rs:178
  17:     0x55a4d82c2b5e - panic_unwind::__rust_maybe_catch_panic
                        at /checkout/src/libpanic_unwind/lib.rs:99
  18:     0x55a4d82ba94b - std::panicking::try<(),closure>
                        at /checkout/src/libstd/panicking.rs:459
                         - std::panic::catch_unwind<closure,()>
                        at /checkout/src/libstd/panic.rs:365
                         - std::rt::lang_start
                        at /checkout/src/libstd/rt.rs:58
  19:     0x55a4d24d5bad - main
  20:     0x7fb673857b44 - __libc_start_main
  21:     0x55a4d248e473 - <unknown>
  22:                0x0 - <unknown>
ERROR:servo: called `Result::unwrap()` on an `Err` value: Io(Error { repr: Os { code: 104, message: "Connection reset by peer" } })
Servo exited with return value 101
(jessie64)nekrad@orion:~/src/moz/servo$ 
```

### PR DESCRIPTION
**Pull request by mrobinson**
This allows servo to use the ExternalScrollId API from WebRender fixing
some issues related to duplicate scroll root ids.

Fixes #17176.
Fixes #19287.
Fixes #19648.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #17176, #19287, and #19648.

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19933)

### CODE REVIEW

File: tests/wpt/mozilla/tests/mozilla/duplicated_scroll_ids.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <link rel="match" href="duplicated_scroll_ids_ref.html">
+        <title>Ensure that content which produces duplicate scroll ideas does not panic</title>
```
Comment (emilio) [inline tests/wpt/mozilla/tests/mozilla/duplicated_scroll_ids.html]: Scroll ideas? :-)

Comment (mrobinson) [reply tests/wpt/mozilla/tests/mozilla/duplicated_scroll_ids.html]: Heh! Okay. I've fixed this typo.

**Review by emilio — APPROVED** [review]
This looks fine to me, feel free to land with r=me unless you think that this needs very careful review from someone more familiar with scrolling.

**Comment by mrobinson** [comment]
The new version of the branch fixes one more small typo. I feel pretty confident about landing this one. Thanks for the review!

**Comment by mrobinson** [comment]
Of course, since I updated the tests, I also need to update the WPT manifests. The latest version of the branch should have proper manifests in place.

**Pull request merged**